app/controllers/AdditionalMetadataClosureStatusController.scala

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import auth.TokenSecurity
+import configuration.KeycloakConfiguration
+import controllers.util.InputNameAndValue
+import controllers.util.MetadataProperty.{clientSideOriginalFilepath, closureType}
+import graphql.codegen.types.{FileFilters, UpdateFileMetadataInput}
+import org.pac4j.play.scala.SecurityComponents
+import play.api.cache.AsyncCacheApi
+import play.api.data.Form
+import play.api.data.Forms.{boolean, mapping}
+import play.api.mvc.{Action, AnyContent, Request, Result}
+import services.{ConsignmentService, CustomMetadataService}
+
+import java.util.UUID
+import javax.inject.Inject
+import scala.concurrent.Future
+import scala.concurrent.duration.DurationInt
+
+class AdditionalMetadataClosureStatusController @Inject()(val consignmentService: ConsignmentService,
+                                                          val customMetadataService: CustomMetadataService,
+                                                          val keycloakConfiguration: KeycloakConfiguration,
+                                                          val controllerComponents: SecurityComponents,
+                                                          val cache: AsyncCacheApi
+                                                         ) extends TokenSecurity {
+
+
+  val closureStatusForm: Form[ClosureStatusFormData] = Form(
+    mapping(
+      "closureStatus" -> boolean
+        .verifying("You must confirm this closure has been approved before continuing.", b => b)
+    )(ClosureStatusFormData.apply)(ClosureStatusFormData.unapply)
+  )
+
+  val closureStatusField: InputNameAndValue = InputNameAndValue("closureStatus", "Yes, I confirm")
+
+
+  def getClosureStatusPage(consignmentId: UUID, fileIds: List[UUID]): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+
+    val filters = Option(FileFilters(None, Option(fileIds), None))
+
+    for {
+      details <- consignmentService.getConsignmentDetails(consignmentId, request.token.bearerAccessToken)
+      consignment <- consignmentService.getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, filters)
+      response <- if (consignment.files.nonEmpty) {
+        val filePaths = consignment.files.flatMap(_.fileMetadata).filter(_.name == clientSideOriginalFilepath).map(_.value)
+        val closureStatus = consignment.files.flatMap(_.fileMetadata).find(_.name == closureType.name).exists(_.value == closureType.value)
+        cache.set(s"$consignmentId-data", (consignment.consignmentReference, filePaths, details.parentFolderId.get), 1.hour)
```

Comment:
Is there a reason for caching this information?

Reply:
I added this because if the user submits the page without confirming the checkbox then it will again call APIs to get that information to show on the page. So, if we cache then it will pick from the cache instead of calling the APIs.
Let me know your opinion on this please.

Reply:
OK. That sounds reasonable